library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -311,6 +313,10 @@ private Slice getFromReaders(Slice key, long timeBucket)
     @Override
     public Slice get(Slice key, long timeBucket, ReadSource readSource)
     {
+      if (key.getClass() == Slice.class) {
+        //The hashCode of Slice was not correct, so correct it
+        key = new SerializeSlice(key);
+      }
```

Comment:
@vrozov Slice is in netlet, and it's basically just a wrapper for a normal byte array that provides equals, hashCode, offset etc. Is there a reason why we created this Slice class instead of just using ByteBuffer in JDK? https://docs.oracle.com/javase/7/docs/api/java/nio/ByteBuffer.html